content/PDOStatementExecute/PDOStatementExecute.q.md

```diff
@@ -0,0 +1,1600 @@
+
+- On line 4, a comma `,` between the array elements `Anne` and `Frank` is not allowed.
+
+:::
+
+/// type=CR, answer=[tests/PDOStatementExecute/MissingSquareBrackets.php], init=[commands/PDStatementExecute/InsertAnneFrankValue.sql], filename=[connection.php,student.php]
+
+Correct the code so that it outputs the strings `Successfully connected to the database.` and `Successfully inserted values into the table.`
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $data = 'Anne','Frank';
+
+    try {
+        $pstmt = $conn->prepare("INSERT INTO authors (first_name, last_name) VALUES (?,?)");
+        $pstmt->bindParam(1, $data[0], PDO::PARAM_STR);
+        $pstmt->bindParam(2, $data[1], PDO::PARAM_STR);
+        if (!$pstmt->execute()) {
+            throw new Exception("Unable to insert values into the table.");
+        }
+         echo "Successfully inserted values into the table.";
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+
++++
+
+
++++
+
+### Part 4: Practice
+
+/// type=CR, answer=[tests/PDOExecuteStatement/CreatePHPProgramUsingBindValue.php], init=[commands/PDOExecuteStatement/InsertStephenHawkingData.sql], filename=[connection.php,author.php]
+
+Write a PHP program named `author.php` that includes the `connection.php` file to insert an author data into the `authors` table and use the `bindValue()` method to bind a value to a parameter. In `author.php`, add the `require_once()` statement to have the `connection.php` file included. Then, add the `try` and `catch` statements. Inside the `try` block, add a statement that assigns the `prepare()` method call of the `$conn` object which passes the argument `INSERT INTO authors (first_name, last_name) VALUES (?,?)` to the `$pstmt` variable. Then, add the `bindValue()` method call of the `$pstmt` object which passes the argument `(1, 'Stephen', PDO::PARAM_STR)`. Add another `bindValue()` method call of the `$pstmt` object which passes the argument `(2, 'Hawking', PDO::PARAM_STR)`. Next, add the `if` statement that evaluates the negated statement `$pstmt->execute()` inside the parentheses `()`. Inside the `if` block, add a statement that throws an exception message `Unable to insert values into the table.`. After the `if` block, add the `echo` statement that displays the string `Successfully inserted values into the table.`. Then, inside the `catch` block, add the statement `echo $e->getMessage();`. Run the program to view the output. 
```

Comment:
Please rephrase this statement based on the changes in the facts.md content. Thanks.